test/e2e/common/secrets.go

```diff
@@ -313,3 +285,88 @@ func doSecretE2EWithMapping(f *framework.Framework, mode *int32) {
 
 	f.TestContainerOutput("consume secrets", pod, 0, expectedOutput)
 }
+
+//Existence of a second secret with same name and different namespace shouldn't affect the mount
+func doSecretE2EForMountNameSpaceIssue(f *framework.Framework, defaultMode *int32) {
+	var (
+		name            = "secret-test-" + string(uuid.NewUUID())
+		volumeName      = "secret-volume"
+		volumeMountPath = "/etc/secret-volume"
+		namespace1      = f.Namespace
+		namespace2      *api.Namespace
+		err             error
+	)
+
+	if namespace2, err = f.CreateNamespace("secret-namespace", nil); err != nil {
+		framework.Failf("unable to create new namespace %s: %v", namespace2.Name, err)
+	}
+
+	secret1, secret2 := secretForTest(namespace1.Name, name), secretForTest(namespace2.Name, name)
+
+	By(fmt.Sprintf("Creating secrets with name %s and namespaces %s %s", name, namespace1.Name, namespace2.Name))
+
+	if secret1, err = f.ClientSet.Core().Secrets(namespace1.Name).Create(secret1); err != nil {
+		framework.Failf("unable to create test secret %s: %v", secret1.Name, err)
+	}
+
+	if secret2, err = f.ClientSet.Core().Secrets(namespace2.Name).Create(secret2); err != nil {
+		framework.Failf("unable to create test secret %s: %v", secret2.Name, err)
+	}
+
+	secret2.Data = map[string][]byte{
+		"this_should_not_match_previous_info": []byte("similarly_this_should_not_match_previous_info\n"),
+	}
+
```

Comment:
@saad-ali thanks, that is what I was trying to say. Sorry if I wasn't clear. But one more thing:

I think it is not needed to modify `doSecretE2EWithMapping(..)` to take a namespace. It can just take the secret name. That way it will create a secret with the name specified (and no ns specified) and create a pod (again, no ns specified) and use it.

If the secret used is not the one in the default ns (i.e. when no ns is specified) then it will fail (i.e it used the secret created before the function call, in some other ns). If not, it will work and that means we are testing it mounts the secret from the proper ns.

If we do think the default ns is special and it has some merit testing it with some other ns too, then it can be added. Of course. But that might be enough (I think) and it's even easier :-)